### DISCUSSION TITLE
Winget doesn't install anything, blue text, and endless progress wheel on terminal window tab

### DISCUSSION THREAD

**Discussion by jaywatkins** [microsoft/winget-cli#3964] [General]
Hi,
I have a fresh install of Windows 11 Enterprise 23H2 in a VM and _winget_ doesn't work (at all). It just turns the text blue and shows one backslash, then carriage return while keeping an active progress wheel on the terminal window's tab. All available Windows updates have been applied and developer mode is on. I downloaded the latest build from this site and tried to install it. That also didn't work, giving the error: _"This app package is not supported for installation by App Installer because it uses certain restricted capabilities."_ I would really love this tool to work. It could really help streamline this type of process, but like most MSFT software, it don't work (without ample problems, googling and troubleshooting). What else could I try?

**Comment by mdanish-kh** [comment]
What version of winget are you on? (Run `winget -v` to find out) 

This is an issue with older versions (<= 1.2) of WinGet since the CDN for those versions was retired. 

Can you upgrade to the latest version and try again? You can get the latest version from https://aka.ms/getwinget, GitHub Releases or the Microsoft Store

**Reply by jaywatkins** [reply]
I cannot install any later version of the App Installer. Windows 11 says the latest version is already installed, 1.21.3133.0.

**Comment by andrewrdavidson** [comment]
Like OP, fresh install of Windows 11 23H2 (Dec 2023 version) from MSDN media gives blue text spinner for 2 seconds or so, changes prompt colour to blue and exits. No error messages.

**OS Version:** 23H2
**OS Build:** 22631.2861

```
PS C:\> winget -v
v1.2.10691
PS C:\> winget --info
Windows Package Manager v1.2.10691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22631.2861
Package: Microsoft.DesktopAppInstaller v1.17.10691.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
```

So, as mentioned above leads to updating WinGet and following the information here ([https://learn.microsoft.com/en-us/windows/package-manager/winget/](https://learn.microsoft.com/en-us/windows/package-manager/winget/)) on how to install WinGet shows this text:

> You can [get App Installer from the Microsoft Store](https://www.microsoft.com/p/app-installer/9nblggh4nns1#activetab=pivot:overviewtab). If it's already installed, make sure it is updated with the latest version. 

This link just spins and times out with "An error occurred while processing your request.".

Downloading the latest version from GitHub (https://github.com/microsoft/winget-cli/releases/tag/v1.6.3482) to update with the following command `Add-AppxPackage Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle` (in an admin-level PowerShell prompt from where you downloaded the .msix bundle to) successfully updated WinGet and it now works.

@mdanish-kh A few comments: 
1. You might want to check the link in the docs about using Microsoft Store! or check the web server since the link doesn't work.
2. Why retire a CDN when this is still required for the default install/configuration of a up-to-date OS image version? 
3. If 2 cannot be rectified then shouldn't 23H2 be updated with a later (i.e. working) version of WinGet? Since we now cannot use it out of the box and thus making automated builds using WinGet now difficult without hacking in an update to WinGet?

I'd really like to use it, but if you make baffling decisions like this how do you expect us (the end user) to start using it when it just doesn't work out of the box, or provide an easy way other than manual install to get it updated to the latest version?

Apologies for the rant, but you've created a great tool then crippled it. It's frustrating.

**Reply by jbalibrea1** [reply]
You save the day, I install a fresh Windows 11 pro, got the spinner and cant update with microsoft strore. 

Need to download and install like you said! thanks

**Reply by dustojnikhummer** [reply]
This is incredibly annoying as it means I can't use Winget for my post install script. I have to wait for Windows Store to update the package...

**Reply by jaywatkins** [reply]
That can be done separately with PowerShell.

**Reply by andrewrdavidson** [reply]
You all may be interested in this, I've tested it manually and it works but not tried it yet in auto-build scripts

https://github.com/asheroto/winget-install

**Reply by tomsik-radek** [reply]
Wait, on Windows Server? I thought Winget didn't work at all on Server

**Reply by andrewrdavidson** [reply]
Well you can if you install all the pre-reqs. I have used this on Server 2022, and Windows 11 and it works every time :-)

**Reply by denelon** [reply]
You can use the steps explained to target Windows Sandbox:
https://learn.microsoft.com/windows/package-manager/winget/#install-winget-on-windows-sandbox

Server 2019 WinGet is not supported.
Server 2022 WinGet is experimental.
Server 2025 WinGet is in public preview.

**Reply by bobthecowinc** [reply]
> Downloading the latest version from GitHub (https://github.com/microsoft/winget-cli/releases/tag/v1.6.3482) to update with the following command Add-AppxPackage Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle (in an admin-level PowerShell prompt from where you downloaded the .msix bundle to) successfully updated WinGet and it now works.

Thank you so much! It fully works now

**Comment by yllekz** [comment]
Having similar issues with a brand new out-of-the-box system with Win11. Winget is broken out of the gate and none of these separate methods of installing/updating it work. This needs to be prioritized because right now it's not a good look for winget to be this nonfunctional by default.

**Reply by mroche74** [reply]
Same here.  New Windows machine, right out of the box.  Winget command simply turns the text in the console purple, adds a line with a backslash, and does nothing else.

**Reply by denelon** [reply]
WinGet is typically updated via the Microsoft Store after the first login. On some earlier versions of Windows 11 it may take several minutes (~15) to trigger the store update mechanism. That mechanism may also need to update several other components. We've been working through lots of hurdles to get to the point where a fresh Windows install will work with WinGet out of the box, but it will most likely take a few more Windows releases before that's the default state.

Between now and then, you can use the Microsoft.WinGet.Client PowerShell module and run `Repair-WinGetPackageManager -Force -Latest` as the way to get the latest stable release of WinGet installed using a scripted approach. Alternatively, you can download and install the dependencies from the latest GitHub release.

**Comment by tspader** [comment]
Yeah, this is also happening for me. Look, I get that there are reasons that this are happening (like a CDN retirement or a delayed store update), but this just _can't_ happen for a professional piece of software. I'm a developer. I write plenty of bugs, I have shipped and will ship broken stuff, I get it. No one's asking for perfection. But to have the OS' built-in package manager fail, silently, on a completely clean install, leaving absolutely no recourse or even a hint of how to figure out why it's broken is just so completely unacceptable. I'm having trouble communicating how stunned I am that there's an issue from 2023 documenting this issue, and 18 months later it's still happening. 

I encourage anyone stumbling across this issue to stay far away from WinGet. I don't mean to be rude; I'm sure the people working on it are good, honest people. But whatever is going on software wise on this project is indefensible.

**Reply by denelon** [reply]
I get it! I have the exact same frustration myself nearly every time I reimage a device. This is pain I personally experience at least weekly. I wish it were trivial to fix (if it was, it would have already been done). We have been making progress, but clearly, it's not there yet. 

It shouldn't have to be muscle memory to know that I need to install Microsoft.WinGet.Client and run "Repair-WinGetPackageManager" or manually kick off the Microsoft Store updates.

**Comment by HenaNieminen** [comment]
I thought this was only a Tiny11 issue (for VM purposes), but holy crap, it's even a problem on the actual full fat Windows version.

Needless to say, it only speaks volumes of Window's reliability nowdays.

**Comment by shivangrathore** [comment]
Happens with me too, endless loading, just blue text
